importer/trickle/trickledag.go

```diff
@@ -56,3 +58,210 @@ func fillTrickleRec(db *h.DagBuilderHelper, node *h.UnixfsNode, depth int) error
 	}
 	return nil
 }
+
+// TrickleAppend appends the data in `db` to the dag, using the Trickledag format
+func TrickleAppend(base *dag.Node, db *h.DagBuilderHelper) (*dag.Node, error) {
+	// Convert to unixfs node for working with easily
+	ufsn, err := h.NewUnixfsNodeFromDag(base)
+	if err != nil {
+		return nil, err
+	}
+
+	// Get depth of this 'tree'
+	n, j := trickleDepthInfo(ufsn, db.Maxlinks())
+	if n == 0 {
+		// If direct blocks not filled...
+		err := db.FillNodeLayer(ufsn)
+		if err != nil {
+			return nil, err
+		}
+
+		if db.Done() {
+			return ufsn.GetDagNode()
+		}
+
+		// If continuing, our depth has increased by one
+		n++
+	}
+
+	err = appendFillLastChild(ufsn, n-1, j, db)
+	if err != nil {
+		return nil, err
+	}
+
+	// after appendFillLastChild, our depth is now increased by one
+	if !db.Done() {
+		n++
+	}
+
+	// Now, continue filling out tree like normal
+	for i := n; !db.Done(); i++ {
+		for j := 0; j < layerRepeat && !db.Done(); j++ {
+			next := h.NewUnixfsNode()
+			err := fillTrickleRec(db, next, i)
+			if err != nil {
+				return nil, err
+			}
+
+			err = ufsn.AddChild(next, db)
+			if err != nil {
+				return nil, err
+			}
+		}
+	}
+
+	return ufsn.GetDagNode()
+}
+
+func appendFillLastChild(ufsn *h.UnixfsNode, depth int, layerFill int, db *h.DagBuilderHelper) error {
+	if ufsn.NumChildren() > db.Maxlinks() {
```

Comment:
i'd find the opposite more readable: 

```go
if ufsn.NumChildren() <= db.Maxlinks() {
  return nil
}

...
```

because the return really means there's nothing else happening